.circleci/config.yml

```diff
@@ -22,17 +22,22 @@ executors:
 commands:
   set_up_node_modules:
     description: "load node_modules, updating and storing back into cache if necessary"
+    parameters:
+      node_version:
+        description: Which node version we're caching packages for
+        type: integer
+        default: 12
     steps:
       - restore_cache:
           keys:
-            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-node<< parameters.node_version >>-dependencies-{{ checksum "package.json" }}
             # fall back to using the latest cache if no exact match is found
-            - v1-dependencies-
+            - v1-node<< parameters.node_version >>-dependencies-
       - run: yarn install
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
+          key: v1-node<< parameters.node_version >>-dependencies-{{ checksum "package.json" }}
```

Comment:
Should the same fallback be used here, the key without the checksum for package.json?

Reply:
Afaik, Circle allows matching just by key prefix, without having to store it yourself. That's how it was set up before my changes as well.

Reply:
ah okay, so is there need for the top fallback then?

Reply:
Sorry I wasn't being clear. You can specify just the prefix of a key for `restore_cache`. That will take the most recent cache that matches it. This is useful so that when the package.json only changes slightly (update 1-2 packages perhaps) we can let yarn handle the installation of just the difference.

So we don't need to specify the fallback for `save_cache`, but do need to specify it for `restore_cache`.

Reply:
Ah, understood, thank you!